subsys/lorawan/lorawan.c

```diff
@@ -576,7 +576,7 @@ int lorawan_start(void)
 	return 0;
 }
 
-static int lorawan_init(const struct device *dev)
+int lorawan_init(void)
```

Comment:
What is the reasoning behind making applications call this function instead of making the priority user-configurable with Kconfig?

If you stick with this approach, the sample needs to be updated to actually call this.

Reply:
Ok, corrected. That might be a change I did during testing that may not be necessary now.
I'll need to do the tests again just to be sure there is no issue with the initialization order of all the layers needed for the LoRaWAN stack.
The main issue I've found in the in the current implementation is that all the layers are deeply coupled, making the initialization a bit tricky